src/index.js

```diff
@@ -76,92 +72,686 @@ class Radar {
+        { latitude, longitude },
+        (status, addresses) => {
+          callback(status, addresses);
+          return;
+        }
+      );
+    });
+  }
+
+  static reverseGeocodeLocation(
+    {
+      latitude,
+      longitude
+    },
+    callback
+  ) {
+    const publishableKey = Cookie.getCookie(Cookie.PUBLISHABLE_KEY);
+
+    if (!publishableKey) {
+      if (callback) {
+        callback(STATUS.ERROR_PUBLISHABLE_KEY);
+      }
+
+      return;
+    }
+
+    const queryParams = {
+      coordinates: `${latitude},${longitude}`,
+    };
+
+    const host = Cookie.getCookie(Cookie.HOST) || DEFAULT_HOST;
+    const url = `${host}/v1/geocode/reverse`;
+    const method = 'GET';
+    const headers = {
+      Authorization: publishableKey,
+    };
+
+    const onSuccess = (response) => {
+      try {
+        response = JSON.parse(response);
+
+        if (callback) {
+          callback(STATUS.SUCCESS, response.addresses);
+        }
+      } catch (e) {
+        if (callback) {
+          callback(STATUS.ERROR_SERVER);
+        }
+      }
+    };
+
+    const onError = (error) => {
+      if (callback) {
+        callback(error);
+      }
+    };
+
+    Http.request(method, url, queryParams, headers, onSuccess, onError);
+  }
+
+  static geocodeIP(callback) {
+    const publishableKey = Cookie.getCookie(Cookie.PUBLISHABLE_KEY);
+
+    if (!publishableKey) {
+      if (callback) {
+        callback(STATUS.ERROR_PUBLISHABLE_KEY);
+      }
+
+      return;
+    }
+
+    const host = Cookie.getCookie(Cookie.HOST) || DEFAULT_HOST;
+    const url = `${host}/v1/geocode/ip`;
+    const method = 'GET';
+    const headers = {
+      Authorization: publishableKey,
+    };
+
+    const onSuccess = (response) => {
+      try {
+        response = JSON.parse(response);
+
+        if (callback) {
+          callback(STATUS.SUCCESS, response.country);
+        }
+      } catch (e) {
+        if (callback) {
+          callback(STATUS.ERROR_SERVER);
+        }
+      }
+    }
+
+    const onError = (error) => {
+      if (callback) {
+        callback(error);
+      }
+    };
+
+    Http.request(method, url, {}, headers, onSuccess, onError);
+  }
+
+  static getDistance(
+    {
+      destinationLat,
+      destinationLng,
+      modes,
+      units,
+    },
+    callback
+  ) {
+    getCurrentPosition((status, { latitude, longitude }) => {
+      if (status !== STATUS.SUCCESS) {
+        if (callback) {
+          callback(status);
+        }
+        return;
+      }
+
+      this.getDistanceFromLocation(
+        {
+          originLat: latitude,
+          originLng: longitude,
+          destinationLat,
+          destinationLng,
+          modes,
+          units
+        },
+        (status, routes) => {
+          callback(status, routes);
+          return;
+        }
+      );
+    });
+  }
+
+  static getDistanceFromLocation(
+    {
+      originLat,
+      originLng,
+      destinationLat,
+      destinationLng,
```

Comment:
we probably want to make this cleaner - should we have clients send `origin` and `destination` as `lat,lng` pairs. We would then not need the manual construction on L721-L722

Reply:
Yeah, I think so. Let's use the same format as `near`.